floris/simulation/grid.py

```diff
@@ -473,3 +486,38 @@ def set_grid(self) -> None:
     #         xoffset * sind(angle) + yoffset * cosd(angle) + center_of_rotation[1]
     #     )
     #     return rotated_x, rotated_y, self.z
+@define
+class PointsGrid(Grid):
+    """
+    Args:
+        grid_resolution (`Vec3`): The number of grid points to be created in each direction.
+        turbine_coordinates (`list[Vec3]`): The collection of turbine coordinate (`Vec3`) objects.
+        reference_turbine_diameter (:py:obj:`float`): The reference turbine's rotor diameter.
+        grid_resolution (:py:obj:`int`): The number of points on each turbine
+    """
+    points_x: NDArrayFloat = field()
+    points_y: NDArrayFloat = field()
+    points_z: NDArrayFloat = field()
+    x_center_of_rotation: float | None = field(default=None)
```

Comment:
@misi9170 what would it mean if these center of rotation inputs are None? Is that valid?

Reply:
@rafmudaf it is valid, yes---they're only used in `rotate_coordiantes_rel_west`, which handles the case where they are None. However, there is a slight naming inconsistency here: in the `Grid` parent class, these fields are called `xc_rot` and `yc_rot`, so the names here should probably be updated to match.